src/channels/router_channel.erl

```diff
@@ -307,7 +296,13 @@ encode_data(Decoder, #{payload := Payload, port := Port} = TemplateArgs, Channel
                     payload => base64:encode(Payload)
                 })
             )
-    end.
+    end;
+encode_data(_Decoder, TemplateArgs, Channel) ->
+    lager:info("missing payload and port [device: ~p]", [?MODULE:device_id(Channel)]),
+    router_channel_utils:maybe_apply_template(
```

Comment:
I wonder if we should even apply template?

Reply:
That's a fair point. They still might be trying to grab `payload` or the likes and will fail on that.